### ISSUE TITLE
Consider having a developer-specific page (or pages) in the docs

### ISSUE THREAD

**Issue by lkatalin** [sigstore/docs#117]
**Description**
It is important to have good documentation for developers, but not everyone will need to know those details meant for developers only. Should we start a development-specific page or pages within the Sigstore documentation? And what could this look like? Some options:

1. Have a developer-specific page under each project, ex. Rekor -> development, Cosign -> development (on an as-needed basis)
2. Have a developer-specific top section, and add pages under that section for each project (as needed)
3. Keep developer-specific info in the project repos (in README.md files)

Something else?

Personally I think 1 or 3 would make the most sense, because people are likely to look for dev-specific details under each project (rather than at top level), and also because the dev-specific documentation will differ for each project.

Reference issue: https://github.com/sigstore/docs/pull/110

**Comment by ltagliaferri** [comment]
I like the #1 approach. Alternately, we can have the docs in the repos as #3 and then set up automation to pull those files into the frontend site and keep them maintained. That would require more upfront work but less maintenance burden.

**Comment by haydentherapper** [comment]
1 sounds like a good approach

**Comment by Neilblaze** [comment]
Yep, I also like the [#1](https://github.com/sigstore/docs/issues/117#:~:text=Have%20a%20developer%2Dspecific%20page%20under%20each%20project%2C%20ex.%20Rekor%20%2D%3E%20development%2C%20Cosign%20%2D%3E%20development%20(on%20an%20as%2Dneeded%20basis)) approach.
